packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -0,0 +1,185 @@
+      initialItem: _selectedMinute ~/ widget.minuteInterval);
+  }
+
+  Widget _buildHourPicker(BuildContext context) {
+    return new CupertinoPicker(
+      scrollController: hourController,
+      offAxisFraction: -0.15,
+      useMagnifier: _kUseMagnifier,
+      magnification: _kMagnification,
+      itemExtent: _kItemExtent,
+      backgroundColor: _kBackgroundColor,
+      onSelectedItemChanged: (int index) {
+        setState(() {
+          _selectedHour = index;
+          widget.onTimerDurationChanged(
+            new Duration(hours: _selectedHour, minutes: _selectedMinute));
+        });
+      },
+      children: new List<Widget>.generate(24, (int index) {
+        return new Container(
+          alignment: Alignment.center,
+          child: new Text(index.toString().padLeft(2, '  ')),
+        );
+      }),
+    );
+  }
+
+  Widget _buildMinutePicker(BuildContext context) {
+    return new CupertinoPicker(
+      scrollController: minuteController,
+      offAxisFraction: 0.5,
+      useMagnifier: _kUseMagnifier,
+      magnification: _kMagnification,
+      itemExtent: _kItemExtent,
+      backgroundColor: _kBackgroundColor,
+      onSelectedItemChanged: (int index) {
+        setState(() {
+          _selectedMinute = index * widget.minuteInterval;
+          widget.onTimerDurationChanged(
+              new Duration(hours: _selectedHour, minutes: _selectedMinute));
+        });
+      },
+      children: new List<Widget>.generate(60 ~/ widget.minuteInterval, (int index) {
+        final int toMinute = index * widget.minuteInterval;
+        return new Container(
+          alignment: Alignment.centerLeft,
+          child: new Text(toMinute.toString().padLeft(2, '  ')),
+        );
+      }),
+      looping: true,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // There are cases where scrolling lands on invalid entries and the picker has
+    // to automatically scrolls to a valid one. Here we use a scroll end listener
+    // to detect such cases.
+    return new NotificationListener<ScrollEndNotification>(
+      onNotification: (ScrollEndNotification notification) {
+        // Invalid case where both hour and minute values are 0.
+        if (_selectedMinute == 0 && _selectedHour == 0) {
+          minuteController.animateToItem(
+              minuteController.selectedItem + 1,
+              duration: _kAutoScrollDuration,
+              curve: _kAutoScrollCurveStyle);
+        }
+      },
+      child: new Stack(
+        children: <Widget>[
+          new Row(
+            children: <Widget>[
+              new Expanded(child: _buildHourPicker(context)),
+              new Expanded(child: _buildMinutePicker(context)),
+            ],
+          ),
+          new Row(
+            children: <Widget>[
+              new Expanded(
+                child: new Container(
+                  alignment: const Alignment(0.6, 0.0),
+                  child: const Text(
+                    'hours',
```

Comment:
I'm concerned about placing the labels in an overlay on top of the pickers, because changes in the size of the labels should affect the underlying pickers too.

Making sure they're consistent with each other could be complicated.